### ISSUE TITLE
Provide a way to disable consideration of GeneratedCodeAttribute as "generated code"

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#20498]
**Version Used**: 15.3 Preview 3

The analysis to determine if a member is "generated code" as indicated by a `[GeneratedCode]` attribute frequently appears as a prominent time sink in profiling reports when using several analyzers (such as StyleCop Analyzers). Since several analyzer projects have worked to ensure that consideration of this attribute is not a requirement for fully-meaningful results, users should have the ability to disable consideration of this attribute and reclaim the wasted time.

**Comment by sharwell** [comment]
@jasonmalinowski I'm treating this as blocked on having **.editorconfig** support in the compiler.

**Comment by jasonmalinowski** [comment]
@sharwell Yep. I'll add it to the project. Honestly thought we already had this but guess not.
